pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -199,6 +225,70 @@ private void manageRetentionForRealtimeTable(String realtimeTableName, Retention
     }
   }
 
+  @VisibleForTesting
+  void manageRetentionForHybridTable(TableConfig realtimeTableConfig, TableConfig offlineTableConfig) {
+    LOGGER.info("Managing retention for hybrid table: {}", realtimeTableConfig.getTableName());
+    List<String> segmentsToDelete = new ArrayList<>();
+    String realtimeTableName = realtimeTableConfig.getTableName();
+    String rawTableName = TableNameBuilder.extractRawTableName(realtimeTableName);
+    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(rawTableName);
+    try {
+      ZkHelixPropertyStore<ZNRecord> propertyStore = _pinotHelixResourceManager.getPropertyStore();
+      Schema schema = ZKMetadataProvider.getTableSchema(propertyStore, offlineTableName);
+      if (schema == null) {
+        throw new RuntimeException("Failed to get schema for table: " + offlineTableName);
+      }
+      String timeColumn = null;
+      SegmentsValidationAndRetentionConfig validationConfig = offlineTableConfig.getValidationConfig();
+      if (validationConfig != null) {
+        timeColumn = validationConfig.getTimeColumnName();
+      }
+      if (StringUtils.isEmpty(timeColumn)) {
+        throw new RuntimeException("TimeColumn is null or empty for table: " + offlineTableName);
+      }
+      DateTimeFieldSpec dateTimeSpec = schema.getSpecForTimeColumn(timeColumn);
+      if (dateTimeSpec == null) {
+        throw new RuntimeException(String.format("Failed to get DateTimeFieldSpec for time column: %s of table: %s",
+            timeColumn, offlineTableName));
+      }
+      DateTimeFormatSpec timeFormatSpec = dateTimeSpec.getFormatSpec();
+      if (timeFormatSpec.getColumnUnit() == null) {
```

Comment:
This shouldn't be needed. In fact it can break use cases with `SIMPLE_DATE_FORMAT`

Reply:
The `TimeBoundaryInfo` is providing the `timeValue` in the configured time format. We need to convert this value into a value in milliseconds for comparison with `segmentZKMetadata.getEndTimeMs()`. The method `DateTimeFormatSpec.fromFormatToMillis(String dateTimeValue)` appears to handle the `SIMPLE_DATE_FORMAT`. 

Please suggest alternative approach if you see issues with the above approach.

Reply:
The remaining part is correct. We shouldn't check `column unit`